frontend/src/metabase/static-viz/components/XYChart/utils/series.ts

```diff
@@ -50,16 +50,38 @@ export const sortSeries = (series: Series[], type: XAxisType) => {
   });
 };
 
-export const calculateStackedItems = (series: Series[]) => {
+export const calculateStackedItems = (multipleSeries: Series[]) => {
+  const dimensionSeriesIndexMap = multipleSeries.reduce(
```

Comment:
See the detail I wrote in the PR body why this map is needed https://github.com/metabase/metabase/pull/26285#issue-1439670437

Reply:
nit: Maybe if it would be a regular loop it would be easier to read, e.g. 

```
const dimensionSeriesIndexMap: Record<string, Record<number, SeriesDatum>> = {};

multipleSeries.forEach((series, seriesIndex) => {
 ...
});
```

Reply:
Sounds good to me. Fixed.